### ISSUE TITLE
DGT - No comms after "adjourn"

### ISSUE THREAD

**Issue by scottishdeveloper** [lukasmonk/lucaschess#93]
Issue Steps
-----------

1. Start a new game with the DGT board connected 
2. Play the game a little with moves / DGT 
3. Adjourn the game
4. Load lucaschess, resuming the adjourned game (OK)
5. The DGT moves are no longer registered (even though Rabbit Plugin shows the correct "board") (ISSUE)

Observation, DGT only seems to "work" immediately following a new game, but adjourn somehow loses the "state" of the game..

**Comment by lukasmonk** [comment]
If you wish, I can prepare a file that makes a log of all the operations of the DGT, to test the different actions that are carried out in each case.

**Comment by scottishdeveloper** [comment]
Hey Lucas, 

That would be great if you can, I can test and send you a debug log to assist resolving the issue.

Cheers,

Martin

**Comment by lukasmonk** [comment]
You could copy the file (but uncompressed=DGT.py):
[DGT.zip](https://github.com/lukasmonk/lucaschess/files/2687614/DGT.zip)

to the Code folder.

First some moves in a new game, then adjourn, and some other moves.

It creates dgt.log in the root folder, then send it to analyze.

**Comment by scottishdeveloper** [comment]
Attached log file:

[dgt.log](https://github.com/lukasmonk/lucaschess/files/2687744/dgt.log)

1. New game / DGT /
(zoran tourney / human)
e4 d5 exd5 Qxd5 nc3

2. Adjourn

3. Re-opened app, resuming adjourned game (clock appears paused, both sides)

qd6 (several attempts at moving from qd6 to qd5, nothing registered)

Tried to "pause" the "continue" on the UI (clock now ticking), qd6,qd5, nothing resigistered

Position in Rabbit board looks fine (so I think it's sending the correct stuff..)

**Comment by lukasmonk** [comment]
Please do the same test with:
[DGT.zip](https://github.com/lukasmonk/lucaschess/files/2687908/DGT.zip)

I have added more info about lines of code.

**Comment by scottishdeveloper** [comment]
Second log result, I performed exact same procedure / moves as before.

[dgt_2.log](https://github.com/lukasmonk/lucaschess/files/2688224/dgt_2.log)

**Comment by lukasmonk** [comment]
Thanks Martin, something is wrong in the file I sent, it needs some line returns, I have tried to create them automatically, but I can understand the result.
Please could you repeat the same moves with:
[DGT.zip](https://github.com/lukasmonk/lucaschess/files/2696417/DGT.zip)

**Comment by scottishdeveloper** [comment]
Hi Lukas,

I've discovered the issue as I'm writing a chrome plugin for chess.com / dgt.

It seems that when you DGT_WritePosition(..) that is not the "initial position" (eg. when reloading the rabbit plugin/lukaschess), then DGT loses it's "state" of who's turn it is.

I wrote a test app to write an initial position / then try to send it the "next move" (but the Rabbit plugin basically bugs out and "does nothing")

`14633,"19:31.04",438.532,4b64,345,"DllConnection",101,"debugIncoming","writePosition(rnbqkbnr1ppppppp8p11P11PPPP1PPPRNBQKBNR)"
14634,"19:31.04",438.532,4b64,522,"Heart",51,"OnGoalFromDllReceived","Received goal from DLL, position is rnbqkbnr/1ppppppp/8/p7/4P3/8/PPPP1PPP/RNBQKBNR ? KQkq ? 0 0; Is sent to ProgramMoveParser."
14635,"19:31.04",438.533,4b64,781,"ProgramMoveParser::goalPositionReceived",60,"Current position null","Current position is null, cannot find program move."
`
Once I figure it out in my own app, I'll let you know how to update it for Lukaschess.

Cheers,

Martin

**Comment by lukasmonk** [comment]
Thanks Martin.

**Comment by scottishdeveloper** [comment]
Also when using FEN (which has the correct side to move in the notiation) DGT_UseFEN(true), it dosen't work either.

The rabbit plugin is awful :)

**Comment by scottishdeveloper** [comment]
Ok, so, if you start a fresh instance of Rabbit (eg. reload app, from say an adjourned game) - then the "fix" is this (since rabbit is bugged and won't parse current position FEN properly).

1. Adjourn / Stop
2. Ensure that Rabbit (Special commands "Use Start/Stop setup" is enabled)
3. Lift both kings, place the kings down (with the last "king" touching the board being the side to move)
4. Boom, it can then "handle" the adjourned state properly (since it knows which side is to move and the plugin starts "properly" parsing the current game...

Pfff!

So really, the "bug" is with Rabbit (not parsing FEN move order and just stalling on a "non initial position")

The only reason it works from a "new game" is because it knows that position is from move 1, annoying!

Cheers,

Martin

**Comment by lukasmonk** [comment]
Have you consider ask for help to dgt support ? I believe they are very interested to fix all these problems.

**Comment by scottishdeveloper** [comment]
Yes, I think I'll eventually compose an email to them with as much info as I can, so they can hopefully resolve things.

Looks like the last plugin update from them was in 3 May 2015 (32bit) ...

Thanks again for your assistance on this! Will keep you informed.
